src/base/datetime/calendar/header/index.ts

```diff
@@ -0,0 +1,295 @@
+import { html, property, svg, query } from "lit-element";
+import {
+  KucBase,
+  dispatchCustomEvent,
+  CustomEventDetail
+} from "../../../kuc-base";
+import { en, zh, ja } from "../../resource/locale";
+
+type BaseDateTimeCalendarHeaderProps = {
+  language?: "en" | "zh" | "ja";
+  month?: number;
+  year?: number;
+};
+
+export class BaseDateTimeCalendarHeader extends KucBase {
+  @property({ type: String }) language = "en";
+  @property({ type: Number }) month = 1;
+  @property({ type: Number }) year = 2021;
+
+  @query(".kuc-base-datetime-calendar-header__group__year")
+  private _yearSelectEl: HTMLSelectElement | undefined;
+
+  @query(".kuc-base-datetime-calendar-header__group__month")
+  private _monthSelectEl: HTMLSelectElement | undefined;
+
+  constructor(props?: BaseDateTimeCalendarHeaderProps) {
```

Comment:
@dong0 
We don't need support set props on constructor for this part.
We can remove it.
Footer does not contain constructor too:
https://github.com/kintone-labs/kintone-ui-component/blob/3e3349bfe0ce7180a554a53309b398dedba5ee83/src/base/datetime/calendar/footer/index.ts